spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java

```diff
@@ -24,7 +24,7 @@
 import javax.crypto.Mac;
 import javax.crypto.spec.SecretKeySpec;
 
-import org.apache.commons.codec.binary.Base64;
+import org.apache.hc.client5.http.utils.Base64;
```

Comment:
in the branch where we checked the migration with snapshot last week we used a class directly from java: https://github.com/codecentric/spring-boot-admin/compare/master...feat/3.1.x

Reply:
Ah - I see... I was just following the migration guide and exchanged it in favor of Apache HttpClient 5. Both solutions are okay with me.